Assets/VignetteHighlight.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class VignetteHighlight : MonoBehaviour {
+	private static readonly int centerID = Shader.PropertyToID("_Center");
+
+	[SerializeField] private RawImage rawImage;
+	[SerializeField] private float fadeDuration = 1f;
+
+	private new Camera camera = null;
+	private Transform worldTarget = null;
+	private Material material = null;
+	private float baseAlpha = 0f;
+	private float alpha = 0f;
+	private float fadeTime = 0f;
+	private int fadeDirection = 1;
+
+	public Transform WorldTarget { set => worldTarget = value; }
+
+	private void Start() {
+		Canvas canvas = GetComponent<Canvas>();
+
+		if (canvas == null) {
+			Debug.LogError($"{nameof(VignetteHighlight)} requires a canvas");
```

Comment:
You can put a `[RequireComponent(typeof(Canvas))]` attribute on the class to make sure that you can't add the component to a game object without a canvas. (Only checks when adding a component from the editor)